arangod/RocksDBEngine/RocksDBSha256Checksum.h

```diff
@@ -0,0 +1,91 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2021 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "ApplicationFeatures/ApplicationServer.h"
+#include "Basics/Mutex.h"
+
+#include <openssl/evp.h>
+#include <rocksdb/file_checksum.h>
+#include <rocksdb/listener.h>
+
+#include "Logger/LogMacros.h"
+
+
+#include <string>
+
+namespace arangodb {
+
+class RocksDBSha256ChecksumFactory;
+class RocksDBShaFileManager;
+
+class RocksDBSha256Checksum : public rocksdb::FileChecksumGenerator {
+ public:
+  explicit RocksDBSha256Checksum(std::string const& filename, std::shared_ptr<RocksDBShaFileManager> shaFileManager);
+  ~RocksDBSha256Checksum();
+
+  void Update(char const* data, size_t n) override;
+  void Finalize() override;
+
+  std::string GetChecksum() const override;
+
+  char const* Name() const override { return "RocksDBSha256Checksum"; }
```

Comment:
As this name is stored repeatedly in the manifest file, let's try to shorten it here (and in the factory function)
```suggestion
  char const* Name() const override { return "ADBSha256"; }
```